addons/sale_purchase/tests/test_sale_purchase.py

```diff
@@ -305,3 +315,30 @@ def test_pol_custom_attribute(self):
         sale_order.action_confirm()
         pol = sale_order._get_purchase_orders().order_line
         self.assertEqual(pol.name, f"{self.service_purchase_1.display_name}\n\n{product_attribute.name}: {product_attribute_value.name}: {custom_value}")
+
+    def test_pol_analytic_distribution(self):
+        """Confirming SO, analytic account from SO should be set as Analytic Distribution in POL."""
+        self.sale_order_1.analytic_account_id, self.sale_order_2.analytic_account_id = self.test_analytic_account_1, self.test_analytic_account_1
```

Comment:
Is one of these assignments supposed to use the `self.test_analytic_account_2`? If not, then why did you add it to the setup?

Also it's my personal preference that you do this variable assignment on 2 separate lines so it's easier to read (and makes it more obvious that you use the same account for both)

Reply:
Hello @ticodoo 
I modified the analytic distribution for SO2, ensuring that both the analytic distribution and analytic_account_id are distinct for SO2. Additionally, other changes have been implemented.

Thank you :slightly_smiling_face: